### PR TITLE
qnvm_ioblock: add hal driver

### LINKED ISSUE #2: qnvm: add wrapper driver to expose hal_ioblock interface

### PR DESCRIPTION
**Pull request by lilvinz**
* fixes #2

### CODE REVIEW

File: hal/src/qnvm_ioblock.c [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+ *
+ * @api
+ */
+bool nvmioblockDisconnect(NVMIOBlockDriver* nvmioblockp)
+{
+    (void)nvmioblockp;
+    return HAL_SUCCESS;
+}
+
+#endif /* HAL_USE_NVM_MEMORY */
```
Comment (janfietz) [inline hal/src/qnvm_ioblock.c]: Comment of closing if has to be HAL_USE_NVM_IOBLOCK

File: hal/src/qnvm_ioblock.c [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+ * @return                  The media presence status.
+ * @retval true             media is inserted.
+ * @retval false            media is not inserted.
+ *
+ * @api
+ */
+bool nvmioblockIsInserted(NVMIOBlockDriver* nvmioblockp)
+{
+    (void)nvmioblockp;
+    return true;
```
Comment (janfietz) [inline hal/src/qnvm_ioblock.c]: Maybe it is better to let the target decide when an io block is inserted.

File: hal/src/qnvm_ioblock.c [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+ * @return                  The media write protection status.
+ * @retval true             media is write protected.
+ * @retval false            media is not write protected.
+ *
+ * @api
+ */
+bool nvmioblockIsProtected(NVMIOBlockDriver* nvmioblockp)
+{
+    (void)nvmioblockp;
+    return false;
```
Comment (janfietz) [inline hal/src/qnvm_ioblock.c]: Maybe it is better to let the target decide when an io block is protected.

File: hal/src/qnvm_ioblock.c [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+ * @return                  The operation status.
+ * @retval HAL_SUCCESS      the operation succeeded.
+ * @retval HAL_FAILED       the operation failed.
+ *
+ * @api
+ */
+bool nvmioblockSync(NVMIOBlockDriver* nvmioblockp)
+{
+    osalDbgCheck(nvmioblockp != NULL);
+    /* Verify device status. */
```
Comment (janfietz) [inline hal/src/qnvm_ioblock.c]: I do not like these  kind of comments. It is obvious to me what the following debug check does.

**Review by janfietz — COMMENTED** [review]
Overall good work and a great benefit for the project.

**Comment by lilvinz** [comment]
Fixed comments.

**Pull request merged**